modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -0,0 +1,532 @@
+/* global __filename, RTCIceGatherer, RTCIceTransport */
+
+import { getLogger } from 'jitsi-meet-logger';
+import yaeti from 'yaeti';
+import { InvalidStateError } from './errors';
+
+const logger = getLogger(__filename);
+
+// Whether we want to mimic the new Promises based RTCPeerConnection or use the
+// old callbacks based one.
+const USE_PROMISES = false;
+
+/**
+ * RTCPeerConnection shim for ORTC based endpoints (such as Edge).
+ *
+ * The interface is based on the W3C specification of 2015, which matches
+ * the implementation of Chrome nowadays:
+ *
+ *   https://www.w3.org/TR/2015/WD-webrtc-20150210/
+ */
+export default class ortcRTCPeerConnection extends yaeti.EventTarget {
+    /**
+     */
+    constructor(pcConfig) {
+        super();
+
+        logger.debug('constructor() pcConfig:', pcConfig);
+
+        // Closed flag.
+        this._closed = false;
+
+        // Local RTCSessionDescription.
+        this._localDescription = null;
+
+        // Remote RTCSessionDescription.
+        this._remoteDescription = null;
+
+        // Set of local MediaStreams.
+        this._localStreams = new Set();
+
+        // Set of remote MediaStreams.
+        this._remoteStreams = new Set();
+
+        // RTCPeerConnection signalingState.
+        this._signalingState = 'stable';
+
+        // RTCPeerConnection iceGatheringState.
+        // NOTE: This should not be needed, but Edge does not implement
+        // iceGatherer.state.
+        this._iceGatheringState = 'new';
+
+        // Create a RTCIceGatherer.
+        this._iceGatherer = this._createIceGatherer(pcConfig);
+
+        // Create a RTCIceTransport.
+        this._iceTransport = this._createIceTransport(this._iceGatherer);
+    }
+
+    /**
+     */
+    get signalingState() {
+        return this._signalingState;
+    }
+
+    /**
+     */
+    get iceGatheringState() {
+        return this._iceGatheringState;
+    }
+
+    /**
+     */
+    get iceConnectionState() {
+        return this._iceTransport.state;
+    }
+
+    /**
+     */
+    get localDescription() {
+        return this._localDescription;
+    }
+
+    /**
+     */
+    get remoteDescription() {
+        return this._remoteDescription;
+    }
+
+    /**
+     */
+    close() {
+        if (this._closed) {
+            return;
+        }
+
+        this._closed = true;
+
+        logger.debug('close()');
+
+        this._updateAndEmitSignalingStateChange('closed');
+
+        // Close iceGatherer.
+        // NOTE: Not implemented by Edge.
```

Comment:
https://developer.microsoft.com/en-us/microsoft-edge/platform/catalog/?page=1&q=RTCIceGatherer

Edge does not "yet" implement it, but it may do in the future. Once they do, this code will still work. Same applies to other APIs.